### PR TITLE
removes unused libraries and flakey unnecessary spec from mimir

### LINKED ISSUE #2296: test failure in `quasar.precog.common.security.AccessControlSpec`
```
[info] ExamplesSpec
[error]   x support providers delegating services to addons
[error]    the value is false (AccessControlSpec.scala:311)
```

### PR DESCRIPTION
**Pull request by alissapajer**
Closes #2296

### CODE REVIEW

File: mimir/src/test/scala/quasar/mimir/JoinOptimizerSpecs.scala [hunk 1]
```diff
@@ -388,7 +388,7 @@ trait JoinOptimizerSpecs[M[+_]] extends EvaluatorSpecification[M]
 
       val input =
         Filter(IdentitySort,
-          dag.Morph1(IndexedRank,
+          dag.Morph1(Flatten,
```
Comment (djspiewak) [inline mimir/src/test/scala/quasar/mimir/JoinOptimizerSpecs.scala]: Can actually just nuke `JoinOptimizerSpecs` altogether.  Don't need to worry about it for this PR, but in future, it should be on the chopping block.

File: mimir/src/test/scala/quasar/mimir/TypeInferencerSpecs.scala [hunk 2]
```diff
@@ -180,16 +180,15 @@ trait TypeInferencerSpecs[M[+_]] extends EvaluatorSpecification[M]
       val line = Line(1, 1, "")
 
       val input =
-        Morph1(Median,
+        Morph1(toUpperCase,
```
Comment (djspiewak) [inline mimir/src/test/scala/quasar/mimir/TypeInferencerSpecs.scala]: Same with `TypeInferencerSpecs`.

**Comment by djspiewak** [comment]
Conflicts, I assume generated by merging #2423.

**Comment by alissapajer** [comment]
@djspiewak ready for another look.

**Pull request merged**